File: Java/HashTable/BJAlgo_1764.java
import java.io.*;
import java.util.*;

public class BJAlgo_1764 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] nm = br.readLine().split(" ");
        int N = Integer.parseInt(nm[0]);
        int M = Integer.parseInt(nm[1]);

        List result = new ArrayList();

        Map<String, String> member = new HashMap<>();
        for (int i = 0; i < N; i++) {
            String mb = br.readLine();
            member.put(mb, mb);
        }

        for (int i = 0; i < M; i++) {
            String mb = br.readLine();
            if (member.containsKey(mb)) {
                result.add(member.get(mb));
            }
        }

        Collections.sort(result);
        System.out.println(result.size());
        for (int i = 0; i < result.size(); i++) {
            System.out.println(result.get(i));
        }
    }
}

File: Python/HashTable/BJAlgo_1764.py
from sys import stdin
N, M = map(int, stdin.readline().split())

members = {}  # 처음 들어오는 듣도 못한 사람들만 저장
result = []

for _ in range(N):
    member = stdin.readline().strip('\n')
    members[member] = member

# 보도 못한 사람들이 들어오면 바로 저장된 듣도 못한 사람들과 비교해서 중복되는 사람만 저장
for _ in range(M):
    member = stdin.readline().strip('\n')
    try:
        result.append(members[member])
    except KeyError:
        continue

# 사전 순으로 정렬하기
result = sorted(result)
print(len(result))
for i in range(len(result)):
    print(result[i])